### PR TITLE
LibIDL+LibWeb: Support generating namespaces & port the CSS namespace to IDL

### PR DESCRIPTION
**Pull request by trflynn89**
Generating these also fixes a bug where `window.CSS.supports.length` was errantly 2

Screenshot from 2023-03-15 09-54-45

### CODE REVIEW

**Comment by trflynn89** [comment]
edit: removed a fixme from CSS.idl saying to support partial namespaces. Let's just do the same thing we do with partial interfaces and keep them all together in once IDL file.

File: Userland/Libraries/LibWeb/CSS/CSS.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#include <LibWeb/CSS/Parser/Parser.h>
+#include <LibWeb/CSS/PropertyID.h>
+#include <LibWeb/CSS/Serialize.h>
+
+namespace Web::CSS {
+
+// https://www.w3.org/TR/cssom-1/#dom-css-escape
+WebIDL::ExceptionOr<String> escape(JS::VM& vm, StringView identifier)
+{
+    return TRY_OR_THROW_OOM(vm, serialize_an_identifier(identifier));
```
Comment (linusg) [inline Userland/Libraries/LibWeb/CSS/CSS.cpp:19]: `// The escape(ident) operation must return the result of invoking serialize an identifier of ident.`

File: Userland/Libraries/LibWeb/CSS/CSS.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#include <LibJS/Forward.h>
+#include <LibWeb/WebIDL/ExceptionOr.h>
+
+// https://www.w3.org/TR/cssom-1/#namespacedef-css
+namespace Web::CSS {
+
+WebIDL::ExceptionOr<String> escape(JS::VM&, StringView identifier);
+
+bool supports(JS::VM&, StringView property, StringView value);
+WebIDL::ExceptionOr<bool> supports(JS::VM&, StringView condition_text);
```
Comment (linusg) [inline Userland/Libraries/LibWeb/CSS/CSS.h:21]: passing `StringView`s to IDL-generated function is not something we've usually done before, usually it's `{Deprecated,}String const&`. Seems like something we can start doing at scale, just wanted to point that out.

Comment (trflynn89) [reply Userland/Libraries/LibWeb/CSS/CSS.h:21]: Yeah, the caller has a `DeprecatedString` (by way of CSSOMString), and these functions don't want ownership of the strings at all, so I figured one less thing to have to port to `String`

**Pull request merged**